src/sqlfluff/testing/rules.py

```diff
@@ -59,6 +59,9 @@ def assert_rule_fail_in_sql(code, sql, configs=None, line_numbers=None):
     linted = Linter(config=cfg).lint_string(sql, fix=True)
     lerrs = linted.get_violations()
     print("Errors Found: {0}".format(lerrs))
+    for e in lerrs:
```

Comment:
In reproducing the bug and testing the fix, I noticed that the tests were "eating" the infinite recursion error, i.e. if a test case had a `fail_str`, the infinite recursion was caught and treated as an expected lint error. With this change, any **unexpected** exceptions will automatically cause the test to fail. I think this is the behavior we want.

Reply:
Good catch